src/main/java/seedu/address/model/project/Deadline.java

```diff
@@ -0,0 +1,46 @@
+package seedu.address.model.project;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Represents a Project's deadline.
+ */
+public class Deadline {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Deadlines should be entered in yyyy-mm-dd date format";
+
+    /*
+     * The date must be entered in yyyy-mm-dd or yyyy-m-d
+     */
+    public static final String VALIDATION_REGEX = "^\\d{4}\\-(0?[1-9]|1[012])\\-(0?[1-9]|[12][0-9]|3[01])$";
+
+    private LocalDate projectDeadline;
+
+    /**
+     * Constructs a project Deadline.
+     *
+     * @param deadline A valid project deadline.
+     */
+    public Deadline(String deadline) {
+        requireNonNull(deadline);
+        checkArgument(isValidDeadline(deadline), VALIDATION_REGEX);
```

Comment:
Is there an error here? The second parameter for `checkArgument` seems to be `String errorMessage`, which would be satisfied better by the `MESSAGE_CONSTRAINTS` variable.

Reply:
Yup made the change